### ISSUE TITLE
cluster-assign-roles Test-Laptop Hadoop fails on bcpc-vm1

### ISSUE THREAD

**Issue by pu239ppy** [bloomberg/chef-bach#880]
Multiple failures in HDFS and HBASE related recipes 
````
* ruby_block[acquire_lock_to_restart_hbase_master] action createacquire_restart_lock : unable to 
get_restart_lock_holder : unable to connect to ZooKeeper quorum f-bcpc-vm1.bcpc.example.com:2181
  * bash[format-zk-hdfs-ha] action runznode_exists : Unable to connect to zookeeper quorum f-bcpc-vm1.bcpc.example.com:2181
````

**Comment by pu239ppy** [comment]
According to the comments in `cluster_assig_roles.rb`, the intent is to have the Heads ready, but it does not work as expected, zookeeper is not running on bcpc-vm2 
````
91   #
192   # The Hadoop convergence order is complex.
193   #
194   # 1. All nodes are converged on "Role[Basic]"
195   # 2. Head nodes converged on "Role[BCPC-Hadoop-Head]"
196   # 3. Head nodes converged with complete runlists.
197   # 4. All workers converged with complete run lists.
198   #
199   # "install_bootstrap" handles step #2, cheffing the head nodes with
200   # a partial runlist.  The method name comes from the old C-A-R.sh.
201   #
````

**Comment by pu239ppy** [comment]
Looks like `install_bootstrap` never gets called instead `install_hadoop` just calls
https://github.com/bloomberg/chef-bach/blob/master/cluster_assign_roles.rb#L239
````
    target_head_nodes.each do |node|
      puts "#{node[:fqdn]}: Cheffing head node with full runlist"
      chef_node_with_runlist(node: node, runlist: node[:runlist])
    end
````

**Comment by pu239ppy** [comment]
#881 now ensures that zookeeper will run on both nodes however HDFS still fails
````
  * service[generally run hadoop-hdfs-namenode] action restart

    ================================================================================
    Error executing action `restart` on resource 'service[generally run hadoop-hdfs-namenode]'
    ================================================================================
````

**Comment by pu239ppy** [comment]
````
2017-06-12 12:02:24,293 INFO org.apache.hadoop.hdfs.server.namenode.FSNamesystem: Stopping services started for standby state
2017-06-12 12:02:24,294 WARN org.apache.hadoop.hdfs.server.namenode.ha.EditLogTailer: Edit log tailer interrupted
java.lang.InterruptedException: sleep interrupted
        at java.lang.Thread.sleep(Native Method)
        at org.apache.hadoop.hdfs.server.namenode.ha.EditLogTailer$EditLogTailerThread.doWork(EditLogTailer.java:347)
        at org.apache.hadoop.hdfs.server.namenode.ha.EditLogTailer$EditLogTailerThread.access$200(EditLogTailer.java:284)
        at org.apache.hadoop.hdfs.server.namenode.ha.EditLogTailer$EditLogTailerThread$1.run(EditLogTailer.java:301)
        at java.security.AccessController.doPrivileged(Native Method)
        at javax.security.auth.Subject.doAs(Subject.java:360)
        at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1704)
        at org.apache.hadoop.security.SecurityUtil.doAsLoginUserOrFatal(SecurityUtil.java:447)
        at org.apache.hadoop.hdfs.server.namenode.ha.EditLogTailer$EditLogTailerThread.run(EditLogTailer.java:297)
2017-06-12 12:02:24,400 INFO org.apache.hadoop.hdfs.server.namenode.FSNamesystem: Starting services required for active state
2017-06-12 12:02:24,400 WARN org.apache.hadoop.hdfs.qjournal.client.QuorumJournalManager: Quorum journal URI 'qjournal://f-bcpc-vm1
.bcpc.example.com:8485;f-bcpc-vm2.bcpc.example.com:8485/Test-Laptop' has an even number of Journal Nodes specified. This is not rec
ommended!
2017-06-12 12:02:24,411 INFO org.apache.hadoop.hdfs.qjournal.client.QuorumJournalManager: Starting recovery process for unclosed jo
urnal segments...
2017-06-12 12:02:24,475 FATAL org.apache.hadoop.hdfs.server.namenode.FSEditLog: Error: recoverUnfinalizedSegments failed for requir
ed journal (JournalAndStream(mgr=QJM to [192.168.100.11:8485, 192.168.100.12:8485], stream=null))
org.apache.hadoop.hdfs.qjournal.client.QuorumException: Got too many exceptions to achieve quorum size 2/2. 1 successful responses:
192.168.100.11:8485: lastPromisedEpoch: 0
httpPort: 8480
fromURL: "http://f-bcpc-vm1.bcpc.example.com:8480"

1 exceptions thrown:
192.168.100.12:8485: Journal Storage Directory /disk/1/dfs/jn/Test-Laptop not formatted
        at org.apache.hadoop.hdfs.qjournal.server.Journal.checkFormatted(Journal.java:461)
        at org.apache.hadoop.hdfs.qjournal.server.Journal.getLastPromisedEpoch(Journal.java:244)
        at org.apache.hadoop.hdfs.qjournal.server.JournalNodeRpcServer.getJournalState(JournalNodeRpcServer.java:127)
        at org.apache.hadoop.hdfs.qjournal.protocolPB.QJournalProtocolServerSideTranslatorPB.getJournalState(QJournalProtocolServer
SideTranslatorPB.java:118)
        at org.apache.hadoop.hdfs.qjournal.protocol.QJournalProtocolProtos$QJournalProtocolService$2.callBlockingMethod(QJournalProtocolProtos.java:25415)
````

**Comment by pu239ppy** [comment]
Looks like I needed to CAR bootstrap before CAR Hadoop, this is normally handled automated_install.sh

**Comment by vt0r** [comment]
Should be resolved by #882.
